hcx-core/postgresql-client/src/main/java/org/swasth/postgresql/PostgreSQLClient.java

```diff
@@ -9,39 +9,49 @@
     private String url;
     private String user;
     private String password;
-    private String tableName;
 
-    public PostgreSQLClient(String url, String user, String password, String tableName) {
+    public PostgreSQLClient(String url, String user, String password) {
         this.url = url;
         this.user = user;
         this.password = password;
-        this.tableName = tableName;
     }
 
-    private Connection getConnection() throws ClientException {
+    public Connection getConnection() throws ClientException {
         Connection conn;
         try {
             conn = DriverManager.getConnection(url, user, password);
-            System.out.println("Connected to the PostgreSQL server successfully.");
         } catch (Exception e) {
             throw new ClientException("Error connecting to the PostgreSQL server: " + e.getMessage());
         }
         return conn;
     }
 
-    public void insert(String mid, String payload) throws ClientException, SQLException {
+
+    public boolean execute(String query) throws ClientException, SQLException {
+        Connection connection = getConnection();
+        try {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            connection.setAutoCommit(false);
+            boolean result = preparedStatement.execute();
+            connection.commit();
+            return result;
+        } catch (Exception e) {
+            throw new ClientException("Error while performing database operation: " + e.getMessage());
+        } finally {
+            connection.close();
```

Comment:
close PreparedStatement as well

Reply:
As we are reading data from ResultSet, closing PreparedStatement is resulting in `org.postgresql.util.PSQLException: This ResultSet is closed` exception.